e2e/connect.test.tsx

```diff
@@ -5,49 +5,86 @@
+      expect(connectWallet.textContent).toBe('Connect wallet to swap')
+    })
+    it('prompts for wallet connection in the Toolbar', async () => {
+      expect(toolbar.textContent).toBe('Connecting…')
+      await waitFor(() => expect(toolbar.textContent).not.toBe('Connecting…'))
+      expect(toolbar.textContent).toBe('Connect wallet to swap')
+    })
+    it('expects widget not to be disabled', async () => {
+      expect(tokenSelect).toHaveProperty('disabled', true)
+      await waitFor(() => expect(toolbar.textContent).not.toBe('Connecting…'))
+      expect(tokenSelect).toHaveProperty('disabled', false)
+    })
   })
 
-  describe('with jsonRpcUrlMap', () => {
+  describe('with valid integrator jsonRpcUrlMap', () => {
     it('prompts for wallet connection in the Wallet', async () => {
-      act(() => component.rerender(<SwapWidget tokenList={tokens} jsonRpcUrlMap={{ 1: [hardhat.url] }} />))
-      expect(wallet.hidden).toBeTruthy()
+      component = render(<SwapWidget tokenList={tokens} jsonRpcUrlMap={{ 1: [hardhat.url] }} />)
+      expect(connectWallet.textContent).toBe('Connect wallet to swap')
+      await waitFor(() => expect(toolbar.textContent).not.toBe('Connecting…'))
+      expect(connectWallet.textContent).toBe('Connect wallet to swap')
+    })
+    it('prompts for wallet connection in the Toolbar', async () => {
+      component = render(<SwapWidget tokenList={tokens} jsonRpcUrlMap={{ 1: [hardhat.url] }} />)
       expect(toolbar.textContent).toBe('Connecting…')
-
-      act(async () => await waitFor(() => expect(toolbar.textContent).not.toBe('Connecting…')))
-      expect(wallet.hidden).toBeFalsy()
-      expect(wallet.textContent).toBe('Connect wallet to swap')
-      expect(toolbar.textContent).toBe('Enter an amount')
+      await waitFor(() => expect(toolbar.textContent).not.toBe('Connecting…'))
+      expect(toolbar.textContent).toBe('Connect wallet to swap')
+    })
+    it('expects widget not to be disabled', async () => {
+      component = render(<SwapWidget tokenList={tokens} jsonRpcUrlMap={{ 1: [hardhat.url] }} />)
+      expect(tokenSelect).toHaveProperty('disabled', true)
+      await waitFor(() => expect(toolbar.textContent).not.toBe('Connecting…'))
+      expect(tokenSelect).toHaveProperty('disabled', false)
     })
   })
 
-  describe('with provider', () => {
-    it('does not prompt for wallet connection', async () => {
-      act(() => component.rerender(<SwapWidget tokenList={tokens} provider={hardhat.provider} />))
-      expect(wallet.hidden).toBeTruthy()
+  describe('with integrator wallet provider', () => {
+    it('displays connected account chip', async () => {
+      component = render(<SwapWidget tokenList={tokens} provider={hardhat.provider} />)
+      await waitFor(() => expect(toolbar.textContent).not.toBe('Connecting…'))
+      account = await component.findByTestId('account')
+      expect(account.textContent?.toLowerCase()).toBe(
```

Comment:
Move the value to a separate const. It can be computed as a const at the top of the file, as it's not related to the test:

    const HARDHAT_ACCOUNT_DISPLAY_NAME = ...